origin-isolation/resources/helpers.mjs

```diff
@@ -0,0 +1,46 @@
+export function insertIframe(hostname, header) {
+  const url = new URL("send-origin-isolation-header.py", import.meta.url);
+  url.hostname = hostname;
+
+  if (header !== undefined) {
+    url.searchParams.set("header", header);
+  }
+
+  const iframe = document.createElement("iframe");
+  iframe.src = url.href;
+
+  return new Promise((resolve, reject) => {
+    iframe.onload = () => resolve(iframe.contentWindow);
+    iframe.onerror = () => reject(new Error(`Could not load ${iframe.src}`));
+    document.body.append(iframe);
+  });
+}
+
+// This function is coupled to ./send-origin-isolation-header.py, which ensures
+// that sending such a message will result in a message back.
+export async function sendWasmModule(frameWindow) {
+  frameWindow.postMessage(await createWasmModule(), "*");
+  return waitForMessage();
+}
+
+// This function is coupled to ./send-origin-isolation-header.py, which ensures
+// that sending such a message will result in a message back.
+export async function setBothDocumentDomains(frameWindow) {
+  document.domain = document.domain;
```

Comment:
It would be nice if a future test ensures relevant exceptions are still thrown. I.e., I expect we want this to align with the cross-origin isolated check which returns near the very end of the setter.